### DISCUSSION TITLE
Child Selector Tailwind CSS

### DISCUSSION THREAD

**Discussion by sahil-xyrintech** [tailwindlabs/tailwindcss#9917] [Help] [answered]
This is really cool to use [&_.class]:w-10 and very helpful in inline css but it goes lengthy when i have to apply many tailwind css classes in one parent element.
for example:
```
<div class="[&_.abc]:text-gray-600 [&_.abc]:bg-blue-500 [&_.abc]:font-bold [&_.abc]:text-lg>

<span class="abc"> This is a dummy text1</span>
<span class="abc"> This is a dummy text2</span>
<span class="abc"> This is a dummy text3</span>

</div>
```

Am just trying to find a way where i can write this css
 ` [&_.abc]:text-gray-600 [&_.abc]:bg-blue-500 [&_.abc]:font-bold [&_.abc]:text-lg `
in this way
`[&_.abc]:text-gray-600_bg-blue-500_font-bold_text-lg`

Is there any solution or not?

**Comment by jabed-web-dev** [comment]
```html
<div class="[&:nth-child(2)]:[&_p]:text-red-500">
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p>
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p>
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p>
</div>
```

**Reply by sahil-xyrintech** [reply]
what if i want to give  hover and focus and few other class on child elements effect, do i have to write like this
`class="[&:nth-child(2)]:[&_p]:text-red-500  [&:nth-child(2)]:[&_p]:bg-red-500  [&:nth-child(2)]:[&_p]:hover:text-red-600  [&:nth-child(2)]:[&_p]:font-bold  [&:nth-child(2)]:[&_p]:hover:bg-red-600 "`

and this is the same problem that i mentioned.

**Reply by jabed-web-dev** [reply]
You can read this document https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state

**Reply by jabed-web-dev** [reply]
```html
<div class="[&>.a]:hover:text-red-500">
  <p class="a">Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p>
  <p class="b">Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p>
  <p class="c">Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p>
</div>
```

I understand your problem

**Comment by wongjn** [comment, marked as answer]
You might be looking for variant groups but Tailwind does not support this. [Here's a response to a similar suggestion](https://github.com/tailwindlabs/tailwindcss/discussions/9053#discussioncomment-3350470):

> We get this question so many times, so I would like to redirect you here for more information: https://twitter.com/adamwathan/status/1461519820411789314

---

You could try adding the classes directly to the elements instead of the parent, but I understand there are circumstances where this is not possible.

You could also abstract the arbitrary variant to a plugin via `addVariant` to make it less verbose.

**Reply by En3Tho** [reply]
Yeah. But this tweet doesn't actually answer the question whether this will be possible or not. This is one of the things that immediately makes you wonder "can this be simplier to use"?

As for size argument - I believe it's possible to introduce an html optimization phase and replace selector:(class1,class2,class3) to selector:class1 selector:class2 selector:class3 so I don't find it very convincing. Production app won't be bigger but developer productivity can grow.

Also I wonder if this thing can be opt-in via a plugin.
